src/main/java/io/javalin/core/JavalinServer.kt

```diff
@@ -60,41 +58,41 @@ class JavalinServer {
                 jettyRequest.isHandled = true
             }
         }.apply {
-            this.sessionHandler = jettySessionHandler
+            this.sessionHandler = config.sessionHandler
         }
 
-        val webSocketHandler = ServletContextHandler(nullParent, contextPath).apply {
+        val webSocketHandler = ServletContextHandler(nullParent, config.contextPath).apply {
             addServlet(ServletHolder(javalinWsServlet), "/*")
         }
 
         val notFoundHandler = object : SessionHandler() {
             override fun doHandle(target: String, jettyRequest: Request, request: HttpServletRequest, response: HttpServletResponse) {
-                val msg = "Not found. Request is below context-path (context-path: '$contextPath')"
+                val msg = "Not found. Request is below context-path (context-path: '${config.contextPath}')"
                 response.status = 404
                 ByteArrayInputStream(msg.toByteArray()).copyTo(response.outputStream)
                 response.outputStream.close()
-                Javalin.log.warn("Received a request below context-path (context-path: '$contextPath'). Returned 404.")
+                Javalin.log.warn("Received a request below context-path (context-path: '${config.contextPath}'). Returned 404.")
             }
         }
 
-        jettyServer.apply {
+        config.server!!.apply {
             handler = attachJavalinHandlers(server.handler, HandlerList(httpHandler, webSocketHandler, notFoundHandler))
             connectors = connectors.takeIf { it.isNotEmpty() } ?: arrayOf(ServerConnector(server).apply {
-                this.port = jettyPort
+                this.port = config.port
             })
         }.start()
 
-        jettyServer.connectors.filterIsInstance<ServerConnector>().forEach {
-            Javalin.log.info("Listening on ${it.protocol}://${it.host ?: "localhost"}:${it.localPort}$contextPath")
+        config.server!!.connectors.filterIsInstance<ServerConnector>().forEach {
+            Javalin.log.info("Listening on ${it.protocol}://${it.host ?: "localhost"}:${it.localPort}${config.contextPath}")
         }
 
-        jettyServer.connectors.filter { it !is ServerConnector }.forEach {
+        config.server!!.connectors.filter { it !is ServerConnector }.forEach {
             Javalin.log.info("Binding to: $it")
         }
 
         reEnableJettyLogger()
         started = true
-        jettyPort = (jettyServer.connectors[0] as? ServerConnector)?.localPort ?: -1
+        config.port = (config.server!!.connectors[0] as? ServerConnector)?.localPort ?: -1
     }
 
     private fun disableJettyLogger() = org.eclipse.jetty.util.log.Log.setLog(NoopLogger()) // disable logger before server creation
```

Comment:
At the bottom of this file there's unused method, is it still used, going to be used or not needed anymore?

```java
object JettyUtil {
    @JvmStatic
    fun getSessionHandler(sessionHandlerSupplier: Supplier<SessionHandler>): SessionHandler {
        val sessionHandler = sessionHandlerSupplier.get()
        try {
            sessionHandler.sessionCache?.sessionDataStore?.exists("id-that-does-not-exist")
        } catch (e: Exception) {
            // TODO: This should throw... Find a way to check this that doesn't fail for valid SessionHandlers.
            Javalin.log.warn("Failed to look up ID in sessionDataStore. SessionHandler might be misconfigured.")
        }
        return sessionHandler
    }
}
```

Reply:
It's supposed to be called in the server-config, thanks!